alica_engine/src/engine/RunningPlan.cpp

```diff
@@ -395,7 +405,10 @@ void RunningPlan::accept(IPlanTreeVisitor* vis)
     vis->visit(*this);
 
     for (RunningPlan* child : _children) {
-        child->accept(vis);
+        assert(!child->isRetired());
+        if (!child->isRetired()) {
```

Comment:
seems redundant

Reply:
not if the assert is compiled out or continue'd upon